File: 202. Happy Number/isHappy.py
class isHappy:
    def check(self, n, happySet):
        if n in happySet:
            return False
        happySum = 0
        happySet.add(n)
        while n > 0:
            x = n % 10
            n = n // 10
            happySum = happySum + x * x
        if happySum == 1:
            return True
        return self.check(happySum, happySet)

    def solution(self, n: int) -> bool:
        happySet = set()
        return self.check(n, happySet)

File: 202. Happy Number/isHappy.java
class isHappy {
    public boolean solution(int n) {
        Set<Integer> happySet = new HashSet();
        while (n != 1) {
            if (!happySet.add(n)) {
                break;
            }
            int happySum = 0;
            while (n > 0) {
                happySum = happySum + (n % 10) * (n % 10);
                n = n / 10;
            }
            n = happySum;
        }
        return n == 1 ? true : false;
    }
}